### PR TITLE
NEW SIQ test case with new ID: 3

### PR DESCRIPTION
**Pull request by gilschieber**

### CODE REVIEW

**Comment by kate-harrison** [comment]
James, can you and your team please take a look at this PR? We don't have many days left before the end of February :)

**Comment by janetliuboston** [comment]
Yes, I will review it soon.

Thanks,
Janet

On Feb 23, 2018, at 8:55 PM, Kate <notifications@github.com<mailto:notifications@github.com>> wrote:


James, can you and your team please take a look at this PR? We don't have many days left before the end of February :)

**Comment by jni2000** [comment]
Yes, we will have multiple reviewers to review this.

File: src/harness/testcases/WINNF_FT_S_SIQ_testcase.py [hunk 1]
```diff
@@ -67,6 +67,211 @@ def test_WINNF_FT_S_SIQ_2(self):
...
+    fss['deploymentParam'][0]['installationParam']['longitude'] = -98.9
+    self._sas_admin.InjectFss({'record': fss})
+
+    # Inject GWBL station operating at range 3650 - 3700 MHz and associated
+    # GWPZ, within 150 km of FSS (approx. 100 km from FSS location set above).
+    gwpz = json.load(
+        open(os.path.join('testcases', 'testdata', 'gwpz_record_0.json')))
+    self._sas_admin.InjectWisp(gwpz)
+
+    # Load PAL/PPA data, operating channel 3620 - 3630 MHz.
```
Comment (janetliuboston) [inline src/harness/testcases/WINNF_FT_S_SIQ_testcase.py]: Current TS-0061 seems to be 3600 - 3500 MHz for PAL.
"Admin Test Harness injects PPA Information into SAS UUT with an operating channel F within 3600-3650MHz frequency range."

File: src/harness/testcases/WINNF_FT_S_SIQ_testcase.py [hunk 2]
```diff
@@ -67,6 +67,211 @@ def test_WINNF_FT_S_SIQ_2(self):
...
+    # Move device_5 outside PPA and outside 150kms range of FSS
+    device_5['installationParam']['latitude'] = 38.2
+    device_5['installationParam']['longitude'] = -99.5
+
+    # Register devices - Use same userId as in PAL record for devices part of
+    # the PPA cluster list.
+    device_1['userId'] = pal_user_id
+    device_3['userId'] = pal_user_id
+    devices = [device_1, device_2, device_3, device_4, device_5]
+    cbsd_ids = self.assertRegistered(devices)
```
Comment (janetliuboston) [inline src/harness/testcases/WINNF_FT_S_SIQ_testcase.py]: Test spec has CPAS trigger as step #4 before step #5 register five CBSDs.

File: src/harness/testcases/WINNF_FT_S_SIQ_testcase.py [hunk 3]
```diff
@@ -67,6 +67,211 @@ def test_WINNF_FT_S_SIQ_2(self):
...
+
+    request = {'spectrumInquiryRequest': spectrum_inquiries}
+    response = self._sas.SpectrumInquiry(request)['spectrumInquiryResponse']
+
+    # Check Spectrum Inquiry Response
+    self.assertEqual(len(response), 5)
+
+    # Checking response codes 0 for all inquiries. Checks for specific devices
+    # will be treated in case by case sections below.
+    for i in xrange(5):
```
Comment (janetliuboston) [inline src/harness/testcases/WINNF_FT_S_SIQ_testcase.py]: There are 5 spectrum inquiries, should be xrange(4): 0, 1, 2, 3, 4.

Comment (gilschieber) [reply src/harness/testcases/WINNF_FT_S_SIQ_testcase.py]: xrange(4) would iterate though 0, 1, 2, 3 only.

File: src/harness/testcases/WINNF_FT_S_SIQ_testcase.py (hunk 1 above)
Comment (gilschieber) [reply src/harness/testcases/WINNF_FT_S_SIQ_testcase.py]: The PPA operating channel shall be 'within' 3600-3650 MHz, so I have chosen a 10 MHz channel (3620 to 3630 MHz) within this range. Should be ok I think.

File: src/harness/testcases/WINNF_FT_S_SIQ_testcase.py (hunk 2 above)
Comment (gilschieber) [reply src/harness/testcases/WINNF_FT_S_SIQ_testcase.py]: Test spec order for this testcase is a little bit misleading.
Devices 1 and 3 must be part of the cluster list of the PPA, so I need to add them to the PPA info before injecting zone data. For this, I need to register them to get their cbsdId. So actually steps 3 and 5 are connected and cannot be done separately. That's why step 4 is coming last.

File: src/harness/testcases/WINNF_FT_S_SIQ_testcase.py (hunk 1 above)
Comment (janetliuboston) [reply src/harness/testcases/WINNF_FT_S_SIQ_testcase.py]: Ah yes, I agree. Thanks!

File: src/harness/testcases/WINNF_FT_S_SIQ_testcase.py (hunk 2 above)
Comment (janetliuboston) [reply src/harness/testcases/WINNF_FT_S_SIQ_testcase.py]: That makes sense. Thanks!

File: src/harness/testcases/WINNF_FT_S_SIQ_testcase.py (hunk 3 above)
Comment (janetliuboston) [reply src/harness/testcases/WINNF_FT_S_SIQ_testcase.py]: Oops, yes. Thanks.

**Pull request merged**